active-rfcs/0000-finalize-head-body.md

```diff
@@ -51,29 +52,43 @@ Losing `"as": "document"` parse mode will remove some special parser handling, l
 
 In this design it is more "on you" to write valid HTML. This comes from the reality that an imported component can contain unknown HTML, so the compiler can't implicitly assume anything about what is or is not included included the final parent component template.  See https://github.com/withastro/astro/issues/2022 for examples of when this breaks down today. We can help with some static linting, and runtime warnings if the final HTML output is invalid. However, this RFC acknowledges the reality that already exists in v0.21 that imported components break any assumptions and help that we previously attempted to provide.
 
+## Head Injection Changes
+
+- runtime: will remove current post-build head injection, which involves a fragile `'</head>'` string find-and-replace.
+- compiler: will add a new `head: string` (or: `injectHead(): string`) property to the compiler transform options, which will inject the given HTML string into the bottom of a `<head>` element, if one is return by the compiler.
```

Comment:
I like `injectHead()` rather than `injectStyles()` / `injectLink()` because it gives the runtime the freedom to inject anything it needs into the bottom of the head, in whatever order it decides is best.

Reply:
I don't think this should be part of the compiler options though. I would expect this to be part of the runtime code. Either part of the `$$result` object or an import, but I think this is an implementation detail that doesn't need to be spelled out in the RFC.

Reply:
ah, I thought that we needed it to be in the compiler since `transform()` returns a final string, but maybe I misunderstood. Agreed, happy to finalize this in the implementation since this RFC is mainly focused on expected behavior.